pkg/actions/kv_run_results_iterator.go

```diff
@@ -0,0 +1,78 @@
+package actions
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/treeverse/lakefs/pkg/logging"
+
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+type KVRunResultIterator struct {
+	it       kv.MessageIterator
+	ctx      context.Context
+	branchID string
+	commitID string
+}
+
+func NewKVRunResultIterator(ctx context.Context, store kv.StoreMessage, repositoryID, branchID, commitID, prefix string) *KVRunResultIterator {
+	key := kv.FormatPath(getRunPath(repositoryID), prefix)
+	it, err := store.Scan(ctx, key)
+	if err != nil {
+		return nil
+	}
+	return &KVRunResultIterator{
+		it:       *it,
+		ctx:      ctx,
+		branchID: branchID,
+		commitID: commitID,
+	}
+}
+
+func (i *KVRunResultIterator) Next() bool {
+	entry := kv.NewMessageEntry((&RunResultData{}).ProtoReflect().Type())
+	for i.it.Next() {
```

Comment:
The performance here would be poor. You would always have to scan and parse all runs just to get a few. I realize that we rely on Postgres for easy filtering until now, but we can't read everything all runs every time a user clicks on a `Actions` tab in the UI.

Reply:
The only way to avoid it is by adding the branch id and/or commit id the the key's path. WDYT? It will mean that we will need to read only the key.

Reply:
Do we even use `SetIf` in Actions kv implementation? If not, we'll face the same issues as in #1511

Reply:
I don't really understand how SetIf is connected to the result iterator. Can you explain?